Kernel/Bus/USB/SysFSUSB.cpp

```diff
@@ -162,10 +168,10 @@ UNMAP_AFTER_INIT void SysFSUSBBusDirectory::initialize()
     s_procfs_usb_bus_directory = directory;
 }
 
-NonnullRefPtr<SysFSUSBDeviceInformation> SysFSUSBDeviceInformation::create(USB::Device& device)
+ErrorOr<NonnullRefPtr<SysFSUSBDeviceInformation>> SysFSUSBDeviceInformation::create(USB::Device& device)
 {
-    auto device_name = KString::must_create(String::number(device.address()));
-    return adopt_ref(*new SysFSUSBDeviceInformation(move(device_name), device));
+    auto device_name = TRY(KString::try_create(String::number(device.address())));
+    return adopt_nonnull_ref_or_enomem(new SysFSUSBDeviceInformation(move(device_name), device));
```

Comment:
You need to use the `new (nothrow) ...` syntax when using to let new actually return nullptr on failure instead of aborting

Reply:
Does this happen in the kernel? I thought we had exceptions explicitly disabled?

Reply:
Right, we do have exceptions disabled, which means that instead of `new` throwing on out of memory conditions, it just panics :^) `(nothrow)` just lets it return a null pointer instead